ee/clickhouse/queries/funnels/test/test_funnel.py

```diff
@@ -1035,3 +1036,26 @@ def test_funnel_step_breakdown_limit(self):
         # assert that we give 5 at a time at most and that those values are the most popular ones
         breakdown_vals = sorted([res[0]["breakdown"] for res in result])
         self.assertEqual(["5", "6", "7", "8", "9"], breakdown_vals)
+
+    def test_funnel_cohort_breakdown(self):
+        # This caused some issues with SQL parsing
+        _create_person(distinct_ids=[f"person1"], team_id=self.team.pk, properties={"key": "value"})
+        _create_event(
+            team=self.team, event="sign up", distinct_id=f"person1", properties={}, timestamp="2020-01-02T12:00:00Z",
+        )
+        cohort = Cohort.objects.create(team=self.team, name="test_cohort", groups=[{"properties": {"key": "value"}}])
+        filters = {
+            "events": [{"id": "sign up", "order": 0}, {"id": "play movie", "order": 1}, {"id": "buy", "order": 2},],
+            "insight": INSIGHT_FUNNELS,
+            "date_from": "2020-01-01",
+            "date_to": "2020-01-08",
+            "funnel_window_days": 7,
+            "breakdown_type": "cohort",
+            "breakdown": [cohort.pk],
+        }
+        filter = Filter(data=filters)
+        funnel = ClickhouseFunnel(filter, self.team)
+
+        result = funnel.run()
```

Comment:
Haven't worked with cohorts before, so a bit unsure, but if I understand this correctly,  breakdown by cohort means breaking down by certain groups of people?

So, if I have another user without the `key` property, what happens to them? Do they go into a "other" group, or do they get discarded?

In either case, would be nice to have a second grouping in this test? So there's more than one breakdown list?

Reply:
They should be discarded. Basically this is a lot like running `n` funnel queries (where `n` is the number of cohorts selected) – where each query just has a filter only taking users belonging to cohort `x` into account – and then presenting these `n` results together.